src/smart-switch-engine-decorator.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import {core} from 'kaltura-player-js';
+import {SmartSwitch} from './smart-switch';
+
+const {getLogger, FakeEvent} = core;
+
+class SmartSwitchEngineDecorator implements IEngineDecorator {
+  _plugin: SmartSwitch;
+  _logger: Object;
+  _engine: IEngine;
+  _active: boolean = true;
+
+  constructor(engine: IEngine, plugin: SmartSwitch) {
+    this._engine = engine;
+    this._plugin = plugin;
+    this._logger = getLogger('SmartSwitchEngineDecorator');
+    this._plugin.player.addEventListener(this._plugin.player.Event.PLAYER_RESET, () => {
+      this._active = true;
+      this._logger.debug(`Set decorator as active: ${this._active.toString()}`);
+    });
+  }
+
+  get active(): boolean {
+    return this._active;
+  }
+
+  load(startTime: ?number): Promise<Object> {
+    return new Promise((resolve, reject) => {
+      this._logger.debug('load API');
+      (this._plugin.getCdnBalancerUrl() || Promise.resolve())
+        .then(cdnBalancerUrl => {
+          if (cdnBalancerUrl) {
+            this._logger.debug('Set new engine src', cdnBalancerUrl);
+            this._engine.src = cdnBalancerUrl;
+          }
+        })
+        .catch(err => this._logger.warn(err.message))
+        .finally(() => {
```

Comment:
working on ie11?

Reply:
R u sure? I think u need a polyfill for finally

Reply:
I'm sure.
You also have finally in playkit-js.